apps/files_external/lib/Lib/Storage/Google.php

```diff
@@ -286,7 +286,8 @@ public function opendir($path) {
 						if ($child->getMimeType() === self::MAP) {
 							continue; // No method known to transfer map files, ignore it
 						} elseif ($child->getMimeType() !== self::FOLDER) {
-							$name .= '.'.$this->getGoogleDocExtension($child->getMimeType());
+							$extension = $this->getGoogleDocExtension($child->getMimeType());
+							if (!empty($extension)) $name .= '.'.$extension;
```

Comment:
Due to PHP wonderful behaviour of `empty()`, if I make a file `test.0` (with file extension the character `0`) then this will not work as expected (`$extension` will be "empty"!)

Maybe:
```suggestion
							if ($extension != '')) $name .= '.'.$extension;
```